src/Microsoft.ComponentDetection.Orchestrator/Experiments/ExperimentService.cs

```diff
@@ -108,6 +108,24 @@ private void FilterExperiments(IComponentDetector detector, int count)
         }
     }
 
+    public void RemoveUnwantedExperimentsbyDetectors(IEnumerable<IComponentDetector> detectors)
+    {
+        var experimentsToRemove = new List<IExperimentConfiguration>();
```

Comment:
This allocation isn't necessary. You can just do `var experimentsToRemove = [linq query]`